storage/indexer/src/db_v2.rs

```diff
@@ -0,0 +1,407 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+/// This file is a copy of the file storage/indexer/src/lib.rs.
+/// At the end of the migration to migrate table info mapping
+/// from storage critical path to indexer, the other file will be removed
+/// and this file will be moved to /ecosystem/indexer-grpc/indexer-grpc-table-info.
+use crate::{
+    metadata::{MetadataKey, MetadataValue},
+    schema::{
+        column_families, indexer_metadata::IndexerMetadataSchema, table_info::TableInfoSchema,
+    },
+};
+use anyhow::{bail, Result};
+use aptos_config::config::RocksdbConfig;
+use aptos_logger::info;
+use aptos_rocksdb_options::gen_rocksdb_options;
+use aptos_schemadb::{SchemaBatch, DB};
+use aptos_storage_interface::{state_view::DbStateView, DbReader};
+use aptos_types::{
+    access_path::Path,
+    account_address::AccountAddress,
+    state_store::{
+        state_key::{StateKey, StateKeyInner},
+        table::{TableHandle, TableInfo},
+    },
+    transaction::Version,
+    write_set::{WriteOp, WriteSet},
+};
+use aptos_vm::data_cache::AsMoveResolver;
+use bytes::Bytes;
+use dashmap::{DashMap, DashSet};
+use move_core_types::{
+    ident_str,
+    language_storage::{StructTag, TypeTag},
+    resolver::MoveResolver,
+};
+use move_resource_viewer::{AnnotatedMoveValue, MoveValueAnnotator};
+use std::{
+    collections::{BTreeMap, HashMap},
+    sync::{
+        atomic::{AtomicU64, Ordering},
+        Arc,
+    },
+    time::Duration,
+};
+
+pub const INDEX_ASYNC_V2_DB_NAME: &str = "index_indexer_async_v2_db";
+const TABLE_INFO_RETRY_TIME_MILLIS: u64 = 10;
+
+#[derive(Debug)]
+pub struct IndexerAsyncV2 {
+    db: DB,
+    next_version: AtomicU64,
+    pending_on: DashMap<TableHandle, DashSet<Bytes>>,
+}
+
+impl IndexerAsyncV2 {
```

Comment:
Add some logs here. The old stuff might not have logs but we should add logs in the new stuff.